runtime/tests/third_party/dummy_device.toml

```diff
@@ -1,95 +1,91 @@
-# the device considers "supported" through PennyLane's device API.
-[[operators.gates]]
-native = [
-        "QubitUnitary",
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "MultiRZ",
-        "Hadamard",
-        "S",
-        "T",
-        "CNOT",
-        "SWAP",
-        "CSWAP",
-        "Toffoli",
-        "CY",
-        "CZ",
-        "PhaseShift",
-        "ControlledPhaseShift",
-        "RX",
-        "RY",
-        "RZ",
-        "Rot",
-        "CRX",
-        "CRY",
-        "CRZ",
-        "CRot",
-        "Identity",
-        "IsingXX",
-        "IsingYY",
-        "IsingZZ",
-        "IsingXY",
-]
+PauliX
+PauliY
+PauliZ
+Hadamard
+Hermitian
+Identity
+Projector
+SparseHamiltonian
+Hamiltonian
+Sum
+SProd
+Prod
+Exp
 
+[operators.gates.decomp]
 # Operators that should be decomposed according to the algorithm used
 # by PennyLane's device API.
 # Optional, since gates not listed in this list will typically be decomposed by
 # default, but can be useful to express a deviation from this device's regular
 # strategy in PennyLane.
-decomp = [
-        "SX",
-        "ISWAP",
-        "PSWAP",
-        "SISWAP",
-        "SQISW",
-        "CPhase",
-        "BasisState",
-        "QubitStateVector",
-        "StatePrep",
-        "ControlledQubitUnitary",
-        "DiagonalQubitUnitary",
-        "SingleExcitation",
-        "SingleExcitationPlus",
-        "SingleExcitationMinus",
-        "DoubleExcitation",
-        "DoubleExcitationPlus",
-        "DoubleExcitationMinus",
-        "QubitCarry",
-        "QubitSum",
-        "OrbitalRotation",
-        "QFT",
-        "ECR",
-]
 
+SX
+ISWAP
+PSWAP
+SISWAP
+SQISW
+CPhase
+BasisState
+QubitStateVector
+StatePrep
+ControlledQubitUnitary
+DiagonalQubitUnitary
+SingleExcitation
+SingleExcitationPlus
+SingleExcitationMinus
+DoubleExcitation
+DoubleExcitationPlus
+DoubleExcitationMinus
+QubitCarry
+QubitSum
+OrbitalRotation
+QFT
+ECR
+
+[operators.gates.matrix]
 # Gates which should be translated to QubitUnitary
-matrix = [
-        "MultiControlledX",
-]
+
+MultiControlledX
+
+[operators.gates.native]
+
+QubitUnitary = { controllable = true }
+PauliX = { controllable = true }
+PauliY = { controllable = true }
+PauliZ = { controllable = true }
+MultiRZ = { controllable = true }
+Hadamard = { controllable = true }
+S = { controllable = true }
+T = { controllable = true }
+CNOT = { }
+SWAP = { controllable = true }
+CSWAP = { }
+Toffoli = { controllable = true }
+CY = { }
+CZ = { }
+PhaseShift = { controllable = true }
+ControlledPhaseShift = { }
+RX = { controllable = true }
+RY = { controllable = true }
+RZ = { controllable = true }
+Rot = { controllable = true }
+CRX = { }
+CRY = { }
+CRZ = { }
+CRot = { }
+Identity = { controllable = true }
+IsingXX = { controllable = true }
+IsingYY = { controllable = true }
+IsingZZ = { controllable = true }
+IsingXY = { controllable = true }
 
 [measurement_processes]
 exactshots = [
```

Comment:
We can convert this table to the new format as well

Reply:
@dime10 Unfortunately, the "new format" is invalid. I did not run "make test" and effectively used the unmodified toml in my tests. I noticed this today and discovered that toml parser raises errors on docs like 
``` toml
[table.table]

String1
String2
```

Some alternatives I see are:

``` toml
[[table.table.String1]]
[[table.table.String2]]
```
or
``` toml
[table.table]
String1 = {}
String2 = {}
```
But none looks good enough to me, so I reverted to strings

Reply:
Oh I see, is the issue that keys cannot exist by themselves without a value?

I think the format below might make the most sense then, because having the elements be strings in one location but keys in another is kind of strange. It would also easily allow attaching properties in the future. But I'm happy to explore other options as well.
```
[table.table]
String1 = {}
String2 = {}
```

Reply:
Fine, let it be the empty table version. For reference, I have proposed the change to the upstream [here](https://github.com/toml-lang/toml/issues/1018)